graphsdk/src/main/java/com/microsoft/graph/model/DateOnly.java

```diff
@@ -0,0 +1,89 @@
+package com.microsoft.graph.model;
+
+
+import java.text.ParseException;
+import java.util.Locale;
+
+/**
+ * A timezone-nonspecific date
+ */
+public class DateOnly {
+
+    private final int
+            mYear,
+            mMonth,
+            mDay;
+
+    /**
+     * Constructs a timezone-nonspecific DateOnly
+     *
+     * @param dateStr date string of the form <code>yyyy-mm-dd</code>
+     */
+    public DateOnly(String dateStr) throws ParseException {
+        // break the date up into its constituent parts
+        String[] dateInfo = dateStr.split("-");
+
+        // validate our split datestring
+        if (dateInfo.length != 3) {
+            throw new ParseException("Datestring input didn't match format 'yyyy-mm-dd'", 0);
```

Comment:
Its mean to a helpful hint to developers, instead of trying to figure out the index, include the dateStr in the message so it is easy to side by side it.